pkg/asset/agent/common.go

```diff
@@ -34,6 +36,8 @@ func HivePlatformType(platform types.Platform) hiveext.PlatformType {
 	switch platform.Name() {
 	case baremetal.Name:
 		return hiveext.BareMetalPlatformType
+	case external.Name:
+		return hiveext.ExternalPlatformType
```

Comment:
We need something that will get converted to `oci` by this function: https://github.com/openshift/assisted-service/blob/master/internal/controller/controllers/clusterdeployments_controller.go#L691-L708

That means that:
1. we will have to modify that function. (this is a bit of a weird case, because ZTP does not - and probably will never - support oci as a platform, but we need a way to represent it in this ZTP manifest).
2. in the short term, this is more likely to be `OCI` than `External`, because to support a generic external platform in assisted we'd need to add another field to pass the `platformName` as well.